Sources/ResourceBuilder.swift

```diff
@@ -0,0 +1,229 @@
+        addRoute(method: .delete, middleware: middleware, id: id, respond: respond)
+    }
+
+    public func delete<I: ResourceIdentifier>(middleware: Middleware..., id: I.Type = I.self, respond: (request: Request, id: I, content: StructuredData) throws -> Response) {
+        addRoute(method: .delete, middleware: middleware, id: id, respond: respond)
+    }
+
+    public func delete<I: ResourceIdentifier, T: ContentMappable>(middleware: Middleware..., id: I.Type = I.self, content: T.Type = T.self, respond: (request: Request, id: I, content: T) throws -> Response) {
+        addRoute(method: .delete, middleware: middleware, id: id, content: content, respond: respond)
+    }
+}
+
+extension ResourceBuilder {
+    public func options(middleware: Middleware..., respond: Respond) {
+        addRoute(method: .options, middleware: middleware, respond: respond)
+    }
+
+    public func options(middleware: Middleware..., content: StructuredData.Type, respond: (request: Request, content: StructuredData) throws -> Response) {
+        addRoute(method: .options, middleware: middleware, respond: respond)
+    }
+
+    public func options<T: ContentMappable>(middleware: Middleware..., content: T.Type = T.self, respond: (request: Request, content: T) throws -> Response) {
+        addRoute(method: .options, middleware: middleware, content: content, respond: respond)
+    }
+
+    public func options<I: ResourceIdentifier>(middleware: Middleware..., id: I.Type = I.self, respond: (request: Request, id: I) throws -> Response) {
+        addRoute(method: .options, middleware: middleware, id: id, respond: respond)
+    }
+
+    public func options<I: ResourceIdentifier>(middleware: Middleware..., id: I.Type = I.self, respond: (request: Request, id: I, content: StructuredData) throws -> Response) {
+        addRoute(method: .options, middleware: middleware, id: id, respond: respond)
+    }
+
+    public func options<I: ResourceIdentifier, T: ContentMappable>(middleware: Middleware..., id: I.Type = I.self, content: T.Type = T.self, respond: (request: Request, id: I, content: T) throws -> Response) {
+        addRoute(method: .options, middleware: middleware, id: id, content: content, respond: respond)
+    }
+}
+
+extension ResourceBuilder {
+    public func addRoute(method: Method, middleware: [Middleware], respond: Respond) {
+        addRoute(method: method, path: "", middleware: middleware, responder: BasicResponder(respond))
+    }
+
+    public func addRoute(method: Method, middleware: [Middleware], respond: (request: Request, content: StructuredData) throws -> Response) {
+        let responder = BasicResponder { request in
+            guard let content = request.content else {
+                throw ClientError.badRequest
+            }
+            return try respond(request: request, content: content)
+        }
+        addRoute(method: method, path: "", middleware: middleware, responder: responder)
```

Comment:
We can also check if the final URI will be empty string `""` and make it a slash `"/"`.